### DISCUSSION TITLE
Support Flow/Operation retry

### DISCUSSION THREAD

**Discussion by falmanna** [directus/directus#18577] [Feature Requests]
### Summary

I believe it will be very useful to have the option to configure retry logic for flows and/or operations. 

### Motivation

Flows can be used as a sync/async workflow orchestration tool. And whenever you talk about workflows, you talk about failures. 

### Detailed Design

Add an option to configure complete flows or a single operation to retry on failure. 
Proposed options:
1. Toggle to enable/disable retry
2. Retry delay 
3. Max number of retries

### Must Have:
- Retry options for operations

### Should Have:
- Retry options for flows

### Could Have:
- Continue from retry for existing failed flows

### Drawbacks

Implementing retry logic can potentially increase the requirements for hosting directus and might reduce compatibility with serverless deployments.

### Alternatives

Using external orchestration tools. 

### Adoption Strategy

This should not be a breaking change. It will be disabled by default for existing flows.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Comment by wuservices** [comment]
I agree it would be great to have some built-in options to retry or queue failures.

Without that, I'm thinking something like https://temporal.io/ could help out. If you have custom code in a flow, then you could use another service with that code to handle all of the retries, error handling, etc.

**Reply by falmanna** [reply]
We went with cloud tasks at the moment. We are creating workflows as custom apis and triggering them from directus trigger -> cloud task -> custom api
